shared_model/bindings/model_transaction_builder_embind.cpp

```diff
@@ -0,0 +1,143 @@
+  .function("hash", &UnsignedWrapper::hash)
+  ;
+
+  typedef ModelProto<UnsignedWrapper> ModelProto;
+  class_<ModelProto>("ModelProtoTransaction")
+  .constructor<UnsignedWrapper&>()
+  .function("signAndAddSignature", &ModelProto::signAndAddSignature)
+  .function("finish", &ModelProto::finish)
+  ;
+
+  using Role = interface::permissions::Role;
+  enum_<Role>("Role")
+  .value("kAppendRole", Role::kAppendRole)
+  .value("kCreateRole", Role::kCreateRole)
+  .value("kDetachRole", Role::kDetachRole)
+  .value("kAddAssetQty", Role::kAddAssetQty)
+  .value("kSubtractAssetQty", Role::kSubtractAssetQty)
+  .value("kAddPeer", Role::kAddPeer)
+  .value("kAddSignatory", Role::kAddSignatory)
+  .value("kRemoveSignatory", Role::kRemoveSignatory)
+  .value("kSetQuorum", Role::kSetQuorum)
+  .value("kCreateAccount", Role::kCreateAccount)
+  .value("kSetDetail", Role::kSetDetail)
+  .value("kCreateAsset", Role::kCreateAsset)
+  .value("kTransfer", Role::kTransfer)
+  .value("kReceive", Role::kReceive)
+  .value("kCreateDomain", Role::kCreateDomain)
+  .value("kReadAssets", Role::kReadAssets)
+  .value("kGetRoles", Role::kGetRoles)
+  .value("kGetMyAccount", Role::kGetMyAccount)
+  .value("kGetAllAccounts", Role::kGetAllAccounts)
+  .value("kGetDomainAccounts", Role::kGetDomainAccounts)
+  .value("kGetMySignatories", Role::kGetMySignatories)
+  .value("kGetAllSignatories", Role::kGetAllSignatories)
+  .value("kGetDomainSignatories", Role::kGetDomainSignatories)
+  .value("kGetMyAccAst", Role::kGetMyAccAst)
+  .value("kGetAllAccAst", Role::kGetAllAccAst)
+  .value("kGetDomainAccAst", Role::kGetDomainAccAst)
+  .value("kGetMyAccDetail", Role::kGetMyAccDetail)
+  .value("kGetAllAccDetail", Role::kGetAllAccDetail)
+  .value("kGetDomainAccDetail", Role::kGetDomainAccDetail)
+  .value("kGetMyAccTxs", Role::kGetMyAccTxs)
+  .value("kGetAllAccTxs", Role::kGetAllAccTxs)
+  .value("kGetDomainAccTxs", Role::kGetDomainAccTxs)
+  .value("kGetMyAccAstTxs", Role::kGetMyAccAstTxs)
+  .value("kGetAllAccAstTxs", Role::kGetAllAccAstTxs)
+  .value("kGetDomainAccAstTxs", Role::kGetDomainAccAstTxs)
+  .value("kGetMyTxs", Role::kGetMyTxs)
+  .value("kGetAllTxs", Role::kGetAllTxs)
+  .value("kSetMyQuorum", Role::kSetMyQuorum)
+  .value("kAddMySignatory", Role::kAddMySignatory)
+  .value("kRemoveMySignatory", Role::kRemoveMySignatory)
+  .value("kTransferMyAssets", Role::kTransferMyAssets)
+  .value("kSetMyAccountDetail", Role::kSetMyAccountDetail)
+  .value("kGetBlocks", Role::kGetBlocks)
+  ;
+
+  using RolePermissionSet = interface::RolePermissionSet;
+  class_<RolePermissionSet>("RolePermissionSet")
+  .constructor<>()
+  .function("size", &RolePermissionSet::size)
+  .function("reset", &RolePermissionSet::reset)
+  .function("set", &RolePermissionSet::set)
+  .function("unset", &RolePermissionSet::unset)
+  .function("test", &RolePermissionSet::test)
+  .function("none", &RolePermissionSet::none)
+  .function("isSubsetOf", &RolePermissionSet::isSubsetOf)
+  ;
```

Comment:
What about operators?

Reply:
It's not a preferred or planned task now.
In the next iteration.